go/vt/topo/etcd2topo/server.go

```diff
@@ -71,12 +82,42 @@ func (s *Server) Close() {
 	s.cli = nil
 }
 
-// NewServer returns a new etcdtopo.Server.
-func NewServer(serverAddr, root string) (*Server, error) {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   strings.Split(serverAddr, ","),
+// TODO: Rename this to NewServer and change NewServer to a name that signifies it's global.
+func NewServerWithOpts(serverAddr, root string, certPath *string, keyPath *string, caPath *string) (*Server, error){
```

Comment:
Just pass the strings by value.

Reply:
Done https://github.com/vitessio/vitess/pull/5496/commits/bd796b42ba368043f1582c3ef9d20f9034766dab.